docs/source/dev-guide/deep-dive-install.md

```diff
@@ -0,0 +1,646 @@
+* `neutered_specs`: `MatchSpec` objects that were already in history but had to be relaxed in order
+  to avoid solving conflicts.
+
+Whatever happens after instantiation depends on the content of these `PrefixSetup` objects.
+Sometimes, the transaction results in no actions (see the [`nothing_to_do`][nothing_to_do]
+property) because the request asked by the user is already fulfilled by the current state
+of the environment.
+
+However, most of the time the transaction will involve a number of actions. This is done via two
+public methods:
+
+* `download_and_extract()`: essentially a forwarder to instantiate and call
+  `ProgressiveFetchExtract`, responsible for deciding which `PackageRecords` need to be
+  downloaded and extracted to the packages cache.
+* `execute()`: the core logic is layed out here. It involves preparing, verifying and
+  performing the rest of the actions. Among others:
+    * Unlinking packages (removing a package from the environment)
+    * Linking (adding a package to the environment)
+    * Compiling bytecode (generating the `pyc` counterpart for each `py` module)
+    * Adding entry points (generate command line executables for the configured functions)
+    * Adding the JSON records (for each package, a JSON file is added to `conda-meta/`)
+    * Make menu items (create shortcuts for packages featuring a JSON file under `Menu/`)
+    * Remove menu items (remove the shortcuts created by that package)
+
+It's important to notice that download and extraction happen separately from all the other actions.
+This separation is important and core to the idea of what a `conda` environment is. Essentially,
+when you create a new `conda` environment, you are not necessarily _copying_ files over to the target
+prefix location. Instead, `conda` maintains a cache of every package ever downloaded to disk (both
+the tarball and the extracted contents). To save space and speed up environment creation and
+deletion, files are not copied over, but instead they are linked (usually via a hardlink). That's
+why these two tasks are separated in the transaction logic: you don't need to download and extract
+packages that are already in the cache; you only need to link them!
+
+```{admonition} Transactions also drive reports
+
+The type and number of actions can also be calculated by `_make_legacy_action_groups()`, which
+returns a list of _action groups_ (one per `PrefixSetup`). Each action group is a just a dictionary
+following this specification:
+
+  ```
+  {
+    "FETCH": Iterable[PackageRecord],  # estimated by `ProgressiveFetchExtract`
+    "PREFIX": str,
+    "UNLINK": Iterable[PackageRecord],
+    "LINK: Iterable[PackageRecord],
+  }
+  ```
+
+
+These simpler action groups are only used for reporting, either via a processed text report
+(via `print_transaction_summary`) or just the raw JSON (via `stdout_json_success`). As you can see,
+they do not know anything about other types of tasks.
+```
```

Comment:
The formatting here is wonky, please see screenshot below for reference:

![Screen Shot 2022-02-14 at 9 58 43 PM](https://user-images.githubusercontent.com/28930622/153984384-0041aab8-4324-4033-9fed-61e9ca005cef.png)

It might not be possible to have a code snippet in the middle of an admonition block, unfortunately.

Reply:
Oh, I think I was missing a set of backticks in the outer level.

Reply:
Can you check with the last changes in b906271?

Reply:
Looks like you fixed it! 👍 

![Screen Shot 2022-02-15 at 9 48 29 AM](https://user-images.githubusercontent.com/28930622/154085975-59f99e45-09d4-4c58-99be-6996d82afac8.png)